src/components/Ref/Ref.tsx

```diff
@@ -30,15 +41,39 @@ export default class Ref extends React.Component<RefProps> {
     innerRef: PropTypes.oneOfType([PropTypes.func, PropTypes.object]),
   }
 
+  static getDerivedStateFromProps(props: RefProps) {
+    const child = React.Children.only(props.children)
+
+    return {
+      child,
+      isForward: isForwardRef(child),
+    }
+  }
+
```

Comment:
I'm not happy that we have the `state` there, however without it we will need to duplicate checks in `componentDidMount` and `componentWillUnmount`.

I'm opened for better ideas 🙀

Reply:
extract duplicate code in a private method and call that method from both `componentDidMount` and `componentWillUnmount`?

Reply:
We need check for `isForwardRef(child)` in render, too.